code/modules/factory/outputter.dm

```diff
@@ -0,0 +1,163 @@
+
+
+
+/obj/item/factory_refill
+	name = "generic refiller"
+	desc = "you shouldnt be seeing this."
+	icon = 'icons/obj/factory/factoryparts.dmi'
+	icon_state = "refillbox"
+	///Typepath for the output machine we want to be ejecting
+	var/obj/item/factory_part/refill_type = /obj/item/factory_part
+	///By how much we wan to refill the target machine
+	var/refill_amount = 30
+
+/obj/item/factory_refill/Initialize()
+	. = ..()
+	var/obj/path = initial(refill_type.result)
+	var/matrix/shift = matrix().Scale(0.4,0.4)
+	var/image/result_image = image(initial(path.icon), initial(path.icon_state), pixel_x = 6, pixel_y = -6)
+	result_image.transform = shift
+	add_overlay(result_image)
+
+/obj/item/factory_refill/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	to_chat(user, "It has [refill_amount] packages remaining.")
+
+/obj/machinery/outputter
+	name = "Unboxer"
+	desc = "An industrial resourcing unboxer."
+	icon = 'icons/obj/factory/factory_machines.dmi'
+	icon_state = "unboxer_inactive"
+	resistance_flags = XENO_DAMAGEABLE
+	density = TRUE
+	anchored = FALSE
+	///the amount of resouce we have left to output factory_parts
+	var/production_amount_left = 0
+	///Maximum amount of resource we can hold
+	var/max_fill_amount = 100
+	///Typepath for the result we want outputted
+	var/obj/item/factory_part/production_type = /obj/item/factory_part
+	///Bool for whether the outputter is producing things
+	var/on = FALSE
+
+/obj/machinery/outputter/Initialize()
+	. = ..()
+	add_overlay(image(icon, "direction_arrow"))
+
+/obj/machinery/outputter/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	to_chat(user, "It is currently facing [dir2text(dir)], and is outputting [initial(production_type.name)]. It has [production_amount_left] resources remaining.")
+
+/obj/machinery/outputter/wrench_act(mob/living/user, obj/item/I)
+	anchored = !anchored
+	balloon_alert(user, "[anchored ? "" : "un"]anchored")
+
+/obj/machinery/outputter/screwdriver_act(mob/living/user, obj/item/I)
+	setDir(turn(dir, 90))
+	balloon_alert(user, "Facing [dir2text(dir)]")
+
+/obj/machinery/outputter/update_icon_state()
+	if(datum_flags & DF_ISPROCESSING)
+		icon_state = "unboxer"
+	else
+		icon_state = "unboxer_inactive"
+
+/obj/machinery/outputter/attack_hand(mob/living/user)
+	if(!anchored)
+		balloon_alert(user, "Must be anchored!")
+		return
+	on = !on
+	if(on)
+		START_PROCESSING(SSmachines, src)
+		balloon_alert_to_viewers("turns on!")
+		update_icon()
+	else
+		STOP_PROCESSING(SSmachines, src)
+		balloon_alert_to_viewers("turns off!")
+		update_icon()
```

Comment:
ditto, and update icon can be placed at the end of the proc if you don't do early return

Reply:
fixed